src/lib/containers/HasAccessV2.tsx

```diff
@@ -0,0 +1,321 @@
+import IconSvg, { Icon } from './IconSvg'
+
+export type HasAccessProps = {
+  onHide?: () => void
+  entityId: string
+  entityVersionNumber?: string
+  className?: string
+}
+
+export const GIGABYTE_SIZE = 2 ** 30
+
+export enum FileHandleDownloadTypeEnum {
+  Accessible = 'Accessible',
+  AccessBlockedByRestriction = 'AccessBlockedByRestriction',
+  AccessBlockedByACL = 'AccessBlockedByACL',
+  AccessBlockedToAnonymous = 'AccessBlockedToAnonymous',
+  NoFileHandle = 'NoFileHandle',
+}
+
+const iconConfiguration: Record<
+  FileHandleDownloadTypeEnum,
+  { icon: Icon; color: (theme: SynapseTheme) => string; tooltipText: string }
+> = {
+  [FileHandleDownloadTypeEnum.AccessBlockedToAnonymous]: {
+    icon: 'accessClosed',
+    color: theme => theme.colors.warning,
+    tooltipText: 'You must sign in to access this file.',
+  },
+  [FileHandleDownloadTypeEnum.AccessBlockedByRestriction]: {
+    icon: 'accessClosed',
+    color: theme => theme.colors.warning,
+    tooltipText: 'You must request access to this restricted file.',
+  },
+  [FileHandleDownloadTypeEnum.AccessBlockedByACL]: {
+    icon: 'accessClosed',
+    color: theme => theme.colors.warning,
+    tooltipText: 'You do not have download access for this item.',
+  },
+
+  [FileHandleDownloadTypeEnum.Accessible]: {
+    icon: 'accessOpen',
+    color: theme => theme.colors.success,
+    tooltipText: '',
+  },
+
+  [FileHandleDownloadTypeEnum.NoFileHandle]: {
+    icon: 'accessOpen',
+    color: theme => theme.colors.success,
+    tooltipText: '',
+  },
+}
+
+function AccessIcon(props: { downloadType: FileHandleDownloadTypeEnum }) {
+  const { downloadType } = props
+  if (downloadType) {
+    const configuration = iconConfiguration[downloadType]
+    return (
+      <ThemeContext.Consumer>
+        {theme => (
+          <IconSvg
+            options={{
+              icon: configuration.icon,
+              color: configuration.color(theme),
+              label: configuration.tooltipText,
+            }}
+          />
+        )}
+      </ThemeContext.Consumer>
+    )
+  }
+  // nothing is rendered until downloadType is determined
+  return <></>
+}
+
+/**
+ * Determines whether an Entity is accessible for download, or if it is blocked by the ACL or unmet Access Requirements.
+ *
+ * To make download available, and determine if the file is downloadable via the web, see {@link DirectDownload.tsx}
+ * @param entityId
+ * @param entityVersionNumber
+ * @returns
+ */
+export function useGetFileHandleDownloadType(
+  entityId: string,
+  entityVersionNumber?: string,
+) {
+  const [fileHandleDownloadType, setFileHandleDownloadType] =
+    React.useState<FileHandleDownloadTypeEnum>()
+
+  const { accessToken } = useSynapseContext()
+  const { data: entityBundle, error: entityFetchError } = useGetEntityBundle(
+    entityId,
+    {
+      includeEntity: true,
+      includePermissions: true,
+    },
+    parseInt(entityVersionNumber ?? ''),
+  )
```

Comment:
HasAccess v1 attempted to fetch the file handle to determine permission (and I think, for the original Download List, to determine if the file handle was external). Instead, just fetch the permission object from the entity bundle.